src/pybamm/models/submodels/particle_mechanics/base_mechanics.py

```diff
@@ -35,6 +39,71 @@ def _get_standard_variables(self, l_cr):
         }
         return variables
 
+    def _get_standard_size_distribution_variables(self, l_cr_dist):
+        domain, Domain = self.domain_Domain
+        l_cr_av_dist = pybamm.x_average(l_cr_dist)
+        variables = {
+            f"{Domain} {self.phase_param.phase_name}particle crack length distribution [m]": l_cr_dist,
+            f"X-averaged {domain} {self.phase_param.phase_name}particle crack length distribution [m]": l_cr_av_dist,
+        }
+        return variables
+
+    def _get_mechanical_size_distribution_results(self, variables):
+        domain, Domain = self.domain_Domain
+        phase_name = self.phase_param.phase_name
+        phase_param = self.phase_param
+        c_s_rav = variables[
+            f"R-averaged {domain} {phase_name}particle concentration distribution [mol.m-3]"
+        ]
+        c_s_surf = variables[
+            f"{Domain} {phase_name}particle surface concentration distribution [mol.m-3]"
+        ]
+        T = pybamm.PrimaryBroadcast(
+            variables[f"{Domain} electrode temperature [K]"],
+            [f"{domain} {phase_name}particle size"],
+        )
+        T = pybamm.PrimaryBroadcast(
+            T,
+            [f"{domain} {phase_name}particle"],
+        )
+
+        # use a tangential approximation for omega
+        c_0 = phase_param.c_0
+        R0 = phase_param.R
+        sto = variables[f"{Domain} {phase_name}particle concentration distribution"]
+        Omega = pybamm.r_average(phase_param.Omega(sto, T))
+
+        E0 = pybamm.r_average(phase_param.E(sto, T))
+        nu = phase_param.nu
+        # Ai2019 eq [10]
+        disp_surf = Omega * R0 / 3 * (c_s_rav - c_0)
+        # c0 reference concentration for no deformation
+        # stress evaluated at the surface of the particles
+        # Ai2019 eq [7] with r=R
+        stress_r_surf = pybamm.Scalar(0)
+        # Ai2019 eq [8] with r=R
+        # c_s_rav is already multiplied by 3/R^3 inside r_average
+        stress_t_surf = Omega * E0 * (c_s_rav - c_s_surf) / 3.0 / (1.0 - nu)
```

Comment:
Based on your last test fix, it looks like this is probably right, tests pass for me too
```suggestion
        stress_t_surf = Omega * E0 * (c_s_rav - c_s_surf) * (1.0 - nu) / 3.0
```